contrib/zip/test/test_zip.cc

```diff
@@ -0,0 +1,395 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <filesystem>
+#include <fstream>
+
+#include "contrib/zip/sandboxed.h"
+#include "contrib/zip/utils/utils_zip.h"
+#include "sandboxed_api/util/path.h"
+#include "sandboxed_api/util/status_matchers.h"
+#include "sandboxed_api/util/temp_file.h"
+
+namespace {
+
+using ::sapi::IsOk;
+
+constexpr size_t kDefaultBlockSize = 19059;
+
+class ZipBase : public testing::Test {
+ protected:
+  std::string GetTestFilePath(const std::string& filename);
+  std::string GetTemporaryFile(const std::string& filename);
+  std::streamsize GetStreamSize(std::ifstream& stream);
+
+  absl::StatusOr<std::vector<uint8_t>> ReadFile(const std::string& filename);
+
+  void SetUp() override;
+
+  const char* test_files_dir_;
+  std::string test_path_zip_;
+
+  std::unique_ptr<ZipSapiSandbox> sandbox_;
+};
+
+class ZipMultiFiles
+    : public ZipBase,
+      public testing::WithParamInterface<std::pair<uint64_t, std::string>> {};
+
+void ZipBase::SetUp() {
+  test_files_dir_ = getenv("TEST_FILES_DIR");
+  ASSERT_NE(test_files_dir_, nullptr);
+
+  test_path_zip_ = GetTestFilePath("zip.zip");
+
+  sandbox_ = std::make_unique<ZipSapiSandbox>();
+  ASSERT_THAT(sandbox_->Init(), IsOk());
+}
+
+absl::StatusOr<std::vector<uint8_t>> ZipBase::ReadFile(
+    const std::string& filename) {
+  std::ifstream file(filename, std::ios::binary);
+  if (!file.is_open()) {
+    return absl::UnavailableError("Unable to open file");
+  }
+
+  std::streamsize size = GetStreamSize(file);
+  std::vector<uint8_t> buf(size);
+
+  file.read(reinterpret_cast<char*>(buf.data()), size);
+
+  if (file.gcount() != size) {
+    return absl::UnavailableError("Unable to read data");
+  }
+
+  return buf;
+}
+
+std::string ZipBase::GetTestFilePath(const std::string& filename) {
+  return sapi::file::JoinPath(test_files_dir_, filename);
+}
+
+std::string ZipBase::GetTemporaryFile(const std::string& filename) {
+  absl::StatusOr<std::string> tmp_file =
+      sapi::CreateNamedTempFileAndClose(filename);
+  if (!tmp_file.ok()) {
+    return "";
+  }
+
+  return sapi::file::JoinPath(sapi::file_util::fileops::GetCWD(), *tmp_file);
+}
+
+std::streamsize ZipBase::GetStreamSize(std::ifstream& stream) {
+  stream.seekg(0, std::ios_base::end);
+  std::streamsize ssize = stream.tellg();
+  stream.seekg(0, std::ios_base::beg);
+
+  return ssize;
+}
+
+TEST_F(ZipBase, CheckInit) {
+  LibZip zip(&(*sandbox_), test_path_zip_, 0);
```

Comment:
Writing `sandbox_.get()` is much clearer than `&(*sandbox_)`. Also in the other tests.

Reply:
Fixed.